pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -182,9 +182,6 @@ func (pt PipelineTask) Validate(ctx context.Context) (errs *apis.FieldError) {
 
 func (pt *PipelineTask) validateMatrix(ctx context.Context) (errs *apis.FieldError) {
 	if pt.IsMatrixed() {
-		// This is an alpha feature and will fail validation if it's used in a pipeline spec
-		// when the enable-api-fields feature gate is anything but "alpha".
-		errs = errs.Also(version.ValidateEnabledAPIFields(ctx, "matrix", config.AlphaAPIFields))
```

Comment:
Would we still want the beta validations for matrix in v1beta1?
My understanding for now (without TEP0138 being merged and the API-compatibility policy being updated), we might still want the feature versioning and API version to be decoupled. I wonder if this validation could also be added as the same for v1. <referring @afrittoli 's previous [beta promotion of isolated steps(TEP0029)](https://github.com/tektoncd/pipeline/pull/6922)>

Reply:
As per the discussion in the API WG, just to be consistent, we are adding such validation where `v1beta1` specifications are validated against `enable-api-fields`. Thanks!

Reply:
Thanks @pritidesai !